### PR TITLE
Changed an URL to a binary GnuPG key used by a test case to Ansible's GitHub

### LINKED ISSUE #74424: The ansible.builtin.apt_key module corrupts binary keys before importing them so import always fails
### Summary

When importing a key by the `apt_key `module, it gets corrupted and not imported in the case of a binary key (not a textual “armored” key). Ansible tries to convert string data, but some data is lost on conversion, which results in a GnuPG error. This is a bug because binary data should not be converted using the `to_native` function.

### Issue Type

Bug Report

### Component Name

ansible.builtin.apt_key module

### Ansible Version

```console
2.12.0.dev0 (latest revision from the GitHub repository) and 2.11.0rc2 (from pip install ansible)
```

### Configuration

```console
config not changed
```

### OS / Environment

Ubuntu 18

### Steps to Reproduce

To reproduce this error, create the following files:

Fie `Vagrantfile` file:

```
Vagrant.configure("2") do |config|
  config.vm.base_mac = nil
  config.vm.synced_folder ".", "/vagrant", disabled: false

  config.vm.define "test-host" do |n|
    n.vm.box = "ubuntu/bionic64"
    n.vm.hostname = "test-host"
    n.vm.provision :ansible do |ansible|
      ansible.limit = "all"
      ansible.playbook = "test.yaml"
    end
  end
end
```

The `test.yaml` file:

```
- name: Test Playbook
  hosts: test-host
  become: yes
  tasks:
    - name: Add binary key
      apt_key: url=https://packages.cloud.google.com/apt/doc/apt-key.gpg
      register: apt_key_binary_test

    - name: Validate the results
      assert:
        that:
          - 'apt_key_binary_test.changed is defined'
          - 'apt_key_binary_test.changed'
```

and run `vagrant up`

### Expected Results

Key imported without a problem

### Actual Results

```console
fatal: [master]: FAILED! => {"changed": false, "msg": "Unable to extract key from '-'", "stderr": "gpg: WARNING: no command supplied.  Trying to guess what you mean ...\ngpg: [don't know]: invalid packet (ctb=0a)\n", "stderr_lines": ["gpg: WARNING: no command supplied.  Trying to guess what you mean ...", "gpg: [don't know]: invalid packet (ctb=0a)"], "stdout": "pub:-:2048:1:FEEA9169307EA071:1614614617:1677728521::-:\nuid:::::::::Rapture Automatic Signing Key (cloud-rapture-signing-key-2021-03-01-08_01_09.pub):\nsub:-:2048:1:AA42F36EE8BEEE0E:1614614617::::\npub:-:2048:1:8B57C5C2836F4BEB:1607040606:1670154510::-:\nuid:::::::::gLinux Rapture Automatic Signing Key (//depot/google3/production/borg/cloud-rapture/keys/cloud-rapture-pubkeys/cloud-rapture-signing-key-2020-12-03-16_08_05.pub) <glinux-team@google.com>:\nsub:-:2048:1:48419E688DD52AC0:1607040606::::\n", "stdout_lines": ["pub:-:2048:1:FEEA9169307EA071:1614614617:1677728521::-:", "uid:::::::::Rapture Automatic Signing Key (cloud-rapture-signing-key-2021-03-01-08_01_09.pub):", "sub:-:2048:1:AA42F36EE8BEEE0E:1614614617::::", "pub:-:2048:1:8B57C5C2836F4BEB:1607040606:1670154510::-:", "uid:::::::::gLinux Rapture Automatic Signing Key (//depot/google3/production/borg/cloud-rapture/keys/cloud-rapture-pubkeys/cloud-rapture-signing-key-2020-12-03-16_08_05.pub) <glinux-team@google.com>:", "sub:-:2048:1:48419E688DD52AC0:1607040606::::"]}
```

Relevant file: 
lib/ansible/modules/apt_key.py

@jvantuyl - please review
@Akasurde - please review

Fixed by https://github.com/ansible/ansible/pull/74474

How did I encounter this error?

@fourstepper 

See https://packages.cloud.google.com/apt/doc/ 

The maintainers of this resource have changed the format of the `apt-key.gpg` file from text to binary on April 22nd, 2021. In the other words, they saved the text version of the key under the new name, `apt-key.gpg.asc`, and put the binary key under the old name. Meanwhile, the rest of the files with the .gpg extension remained text. For this reason, before April 22nd, playbooks that link to this URL directly worked and then stopped. You can replace the URL in the playbook with https://packages.cloud.google.com/apt/doc/apt-key.gpg.asc so that your .yaml works on current Ansible versions and not just on the latest pull from Github.

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by maximmasiutin**
##### SUMMARY
Changed an URL to a binary GnuPG key used by test case to a resource controlled by the Ansible developers on GitHub, rather than by a third party (Google) that could replace the key to a non-binary one or delete it altogether, thus invalidating the test.

This pull request augments the commit https://github.com/ansible/ansible/commit/3bc2e7783cfc4920d807e01c6a1d905abb3d3f15 from the pull request https://github.com/ansible/ansible/pull/74474

Fixes https://github.com/ansible/ansible/issues/74424

##### ISSUE TYPE
- Test Pull Request

##### COMPONENT NAME

`test/integration/targets/apt_key`

##### ADDITIONAL INFORMATION
I could not specify this URL in the initial pull request (see the link below) because the test key itself is in the Ansible repository and was not available via an URL before the pull request got accepted.  

https://github.com/ansible/ansible/commit/3bc2e7783cfc4920d807e01c6a1d905abb3d3f15#diff-e9e6aa5eebd120b7e267ffdd19f94080f5d5c5631d63b076789cef310a7bff6c 

I have generated a new key with GnuPG specially for that test purpose, rather than copying it from somewhere, to exclude probable claims from the key's owner that we have used the key in such a way without permission.

The key's subject is "Test Binary Key (Ansible Test Binary Key) <rfc2606.example.invalid@example.org>"

The key URL is https://github.com/ansible/ansible/raw/devel/test/integration/targets/apt_key/samples/apt-key-example-binary.gpg

Update: the URL https://ansible-ci-files.s3.us-east-1.amazonaws.com/test/integration/targets/apt_key/apt-key-example-binary.gpg is used instead of the previous URL.

### CODE REVIEW

**Comment by maximmasiutin** [comment]
@bcoca - thank you for accepting the https://github.com/ansible/ansible/pull/74474

Could you please also accept this one, since it augments the commit. Could not do that in one steps for the reasons explained above.

**Comment by maximmasiutin** [comment]
ready_for_review

File: test/integration/targets/apt_key/tasks/apt_key_binary.yml [hunk 1]
```diff
@@ -1,9 +1,7 @@
 ---
 
 - name: Ensure import of binary key downloaded using URLs works
-  apt_key: url=https://packages.cloud.google.com/apt/doc/apt-key.gpg
-  # replace the above URL to the following, after the pull request is accepted
-  # apt_key: url=https://github.com/ansible/ansible/tree/devel/test/integration/targets/apt_key/samples/apt-key-example-binary.gpg
+  apt_key: url=https://github.com/ansible/ansible/raw/devel/test/integration/targets/apt_key/samples/apt-key-example-binary.gpg
```
Comment (samdoran) [inline test/integration/targets/apt_key/tasks/apt_key_binary.yml]: I put this in S3. Please remove `apt_key/samples/apt-key-example-binary.gpg`.
```suggestion
  apt_key:
    url: https://ansible-ci-files.s3.us-east-1.amazonaws.com/test/integration/targets/apt_key/apt-key-example-binary.gpg
```

**Comment by samdoran** [comment]
@maximmasiutin Thank you for your recent contributions. Could you please make sure to fill out the template when creating issues and PRs? It helps us to a lot.

**Comment by maximmasiutin** [comment]
Thank you! I will be writing the text of the future pull requests according to https://raw.githubusercontent.com/ansible/ansible/devel/.github/PULL_REQUEST_TEMPLATE.md

I will also review the text of pending pull requests whether it needs to be modified according to the above rules.

**Review by samdoran — APPROVED** [review]

**Pull request merged**